src/Illuminate/Database/Schema/ForeignIdColumnDefinition.php

```diff
@@ -37,6 +37,10 @@ public function __construct(Blueprint $blueprint, $attributes = [])
      */
     public function constrained($table = null, $column = 'id', $indexName = null)
     {
+        if (is_null($table) && is_null($this->table)) {
```

Comment:
Why check if both are `null`? And then assign null to something already `null`?

Would this work?

```php
$table ??= $this-table;
```

note: without any if clauses

Reply:
That was a typo from my side.

Good catch and even better suggestion, thanks :)